core/plugins/stack/python/dependencies.ts

```diff
@@ -0,0 +1,48 @@
+import { Context } from "../../../types.ts";
+
+const readDependencyFile = async (context: Context) => {
+  const dependencies: string[] = [];
+
+  for await (const file of context.files.each("**/pyproject.toml")) {
+    const pyproject = await context.files.readToml(file.path);
+
+    const poetryDeps = pyproject.tool?.poetry?.dependencies;
+    if (poetryDeps) {
+      Array.prototype.push.apply(dependencies, Object.keys(poetryDeps));
+    }
+    const poetryDevDeps = pyproject.tool?.poetry["dev-dependencies"];
+    if (poetryDevDeps) {
+      Array.prototype.push.apply(dependencies, Object.keys(poetryDevDeps));
+    }
+  }
+
+  for await (const file of context.files.each("**/Pipfile")) {
+    const pipfile = await context.files.readToml(file.path);
+
+    const pipFileDeps = pipfile.packages;
+    if (pipFileDeps) {
+      Array.prototype.push.apply(dependencies, Object.keys(pipFileDeps));
+    }
+    const pipFileDevDeps = pipfile["dev-packages"];
+    if (pipFileDevDeps) {
+      Array.prototype.push.apply(dependencies, Object.keys(pipFileDevDeps));
+    }
+  }
+
+  for await (const file of context.files.each("**/requirements.txt")) {
+    const requirements = await context.files.readText(file.path);
+    Array.prototype.push.apply(dependencies, requirements.split("\n"));
```

Comment:
In this case the comparison would be by substring, but as pointed out below it can lead to other problems. I was trying to avoid using the regex or some complex parser, but i guess there's no other way

Reply:
Updated to use a modified version of the regex pointed on the [PEP 508](https://www.python.org/dev/peps/pep-0508/#names) the only difference is the matching of the dependency version number that was removed, you can test here https://regex101.com/r/whDTPw/1